extensions/cloudcredentials/cloudcredentials.go

```diff
@@ -23,3 +31,54 @@ type CloudCredential struct {
 	VmwareVsphereConfig          *VmwarevsphereCredentialConfig `json:"vmwarevspherecredentialConfig,omitempty"`
 	UUID                         string                         `json:"uuid,omitempty"`
 }
+
+func LoadCloudCredential(provider string) CloudCredential {
+	var cloudCredential CloudCredential
+	switch {
+
+	case provider == provisioninginput.AWSProviderName.String():
+		var awsCredentialConfig AmazonEC2CredentialConfig
+		config.LoadConfig(AmazonEC2CredentialConfigurationFileKey, &awsCredentialConfig)
+		cloudCredential.AmazonEC2CredentialConfig = &awsCredentialConfig
+		return cloudCredential
+
+	case provider == provisioninginput.AzureProviderName.String():
+		var azureCredentialConfig AzureCredentialConfig
+		config.LoadConfig(AzureCredentialConfigurationFileKey, &azureCredentialConfig)
+		cloudCredential.AzureCredentialConfig = &azureCredentialConfig
+		return cloudCredential
+
+	case provider == provisioninginput.DOProviderName.String():
+		var digitalOceanCredentialConfig DigitalOceanCredentialConfig
+		config.LoadConfig(DigitalOceanCredentialConfigurationFileKey, &digitalOceanCredentialConfig)
+		cloudCredential.DigitalOceanCredentialConfig = &digitalOceanCredentialConfig
+		return cloudCredential
+
+	case provider == provisioninginput.LinodeProviderName.String():
+		var linodeCredentialConfig LinodeCredentialConfig
+		config.LoadConfig(LinodeCredentialConfigurationFileKey, &linodeCredentialConfig)
+		cloudCredential.LinodeCredentialConfig = &linodeCredentialConfig
+		return cloudCredential
+
+	case provider == provisioninginput.HarvesterProviderName.String():
+		var harvesterCredentialConfig HarvesterCredentialConfig
+		config.LoadConfig(HarvesterCredentialConfigurationFileKey, &harvesterCredentialConfig)
+		cloudCredential.HarvesterCredentialConfig = &harvesterCredentialConfig
+		return cloudCredential
+
+	case provider == provisioninginput.VsphereProviderName.String():
+		var vsphereCredentialConfig VmwarevsphereCredentialConfig
+		config.LoadConfig(VmwarevsphereCredentialConfigurationFileKey, &vsphereCredentialConfig)
+		cloudCredential.VmwareVsphereConfig = &vsphereCredentialConfig
+		return cloudCredential
+
+	case provider == provisioninginput.GoogleProviderName.String():
+		var googleCredentialConfig GoogleCredentialConfig
+		config.LoadConfig(GoogleCredentialConfigurationFileKey, &googleCredentialConfig)
+		cloudCredential.GoogleCredentialConfig = &googleCredentialConfig
+		return cloudCredential
+
+	default:
+		panic(fmt.Sprintf("Provider:%v not found", provider))
```

Comment:
Is there a specific reason we are `panic`? Couldn't we just do a `logrus.Errorf` or something and not panic?

Reply:
I think Sam is good here, if the user provides a non-supported value we should panic.